.gitignore

```diff
@@ -149,3 +149,6 @@ GSYMS
 /tools/cub-1.8.0/
 /tools/cub
 /tools/python/
+
+/src/**/CMakeLists.txt
+/build
```

Comment:
I suggest `/build*` Part of the value of cmake is the ability to have multiple coexisting builds.

pytorch acts similarly: https://github.com/pytorch/pytorch/blob/455b5c1a7d1503c03853e37b5a2caf60948359eb/.gitignore#L186